### PR TITLE
Replace univalue subtree with proper dependency on external UniValue

### PR DESCRIPTION
**Pull request by luke-jr**
This is not consensus-critical code, so really had no excuse to be subtree'd in the first place.

### CODE REVIEW

**Comment by luke-jr** [comment]
Actually, this loses a bugfix, so @jgarzik needs to release UniValue 1.0.2 to make this sane...

**Comment by luke-jr** [comment]
Also, I believe the [Travis failure crash](https://travis-ci.org/bitcoin/bitcoin/jobs/102242679) is unrelated to this PR, and a bug in UniValue (or Core) that we need to fix independently of this - but I can't reproduce it... :/

File: depends/packages/univalue.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package=univalue
+$(package)_version=1.0.1
+$(package)_download_path=https://codeload.github.com/jgarzik/$(package)/tar.gz
```
Comment (jonasschnelli) [inline depends/packages/univalue.mk]: I think we should use the bitcoin/univalue.git repository to allow multiple people to maintain the repository (the subtree is also pointing to bitcoin/univalue.git).

**Comment by jonasschnelli** [comment]
Hmm.. not sure about that.
If I'm right, this would mean, in order to compile bitcoin-core, users need to checkout the univalue git repository and compile univalue by themselves unless there is broad support over package managers (apt-get, etc.)?

IMO we should use subtrees for dependencies that are not available over common package managers (bdb4.8 might be an exception because a) not really necessary [--with-incompatible-bdb] and b) wallet only).

**Comment by laanwj** [comment]
NACK. Univalue is a nice and small library, and it's useful to have it included..
I'd be OK with the *option* to use an external univalue, but don't remove the internal one.

File: depends/packages/univalue.mk (hunk 1 above)
Comment (luke-jr) [inline depends/packages/univalue.mk]: AFAIK that's just a mirror of @jgarzik 's upstream? Note he could very well add additional maintainers to his upstream repo as well, if he wants.

**Comment by luke-jr** [comment]
@laanwj Is it really so much harder to run:

    make -C depends univalue

**Comment by MarcoFalke** [comment]
> make -C depends univalue

You'd have to update `doc/build*`

**Comment by luke-jr** [comment]
Apparently the depends system can't actually be used in the way I expected.

While I still thing this is the correct way to go, I'll start with making it an option for now... in #7349

**Pull request closed without merging**